index.js

```diff
@@ -12,7 +12,26 @@ function isNegative(pattern) {
 	return pattern[0] === '!';
 }
 
+function isString(value) {
+	return typeof value === 'string';
+}
+
+function validatePatternsArray(patterns) {
+	if (isString(patterns)) {
+		// bail early, the rest will easier this way
+		return;
+	}
+
+	if (!Array.isArray(patterns) || !patterns.every(isString)) {
+		throw new TypeError('patterns must be a string or an array of strings');
+	}
+}
+
 function generateGlobTasks(patterns, opts) {
+	// make sure input is correct, so that we can safely parse
+	// it in this task
+	validatePatternsArray(patterns);
+
 	var globTasks = [];
 
 	patterns = arrify(patterns);
```

Comment:
@catdad What do you think about, changing `validatePatternsArray` to `assertPatternsArray` (or similar) and:

```js
function generateGlobTasks(patterns, opts) {
  	var globTasks = [];
  
  	patterns = arrify(patterns);
	assertPatternsArray(patterns);
```

The benefits I see:
1. [assert](https://nodejs.org/api/assert.html) is a well-known term, known to throw errors
2. There is no need anymore to check `isString(patterns)` in `validatePatternsArray`/`assertPatternsArray`